.grit/patterns/java/correct_true_comparison_operator.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Correct comparison operator `$x = true`
+tags: [good-practice]
+---
+
+Update redundant incorrect comparison `($x == true)` to achieve clearer code logic and avoid unnecessary repetition.
```

Comment:
This will be used as an error message, so precision is helpful.

```suggestion
Assignment inside a condition is usually accidental, this is likely meant to be a comparison.
```